File: Java/app/src/main/java/dev/ogabek/java/activity/MainActivity.java
package dev.ogabek.java.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

import dev.ogabek.java.R;
import dev.ogabek.java.adapter.ChatAdapter;
import dev.ogabek.java.model.Chat;
import dev.ogabek.java.model.Message;
import dev.ogabek.java.model.Room;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

    }

    private void initViews() {
        recyclerView = findViewById(R.id.rv_main);
        recyclerView.setLayoutManager(new GridLayoutManager(this, 1));

        refreshAdapter(getAllChats());

    }

    private void refreshAdapter(List<Chat> chats) {
        ChatAdapter adapter = new ChatAdapter(this, getAllChats());
        recyclerView.setAdapter(adapter);
    }

    private ArrayList<Chat> getAllChats() {
        ArrayList<Room> stories = new ArrayList<>();
        stories.add(new Room(R.drawable.aziz, "Azizbek"));
        stories.add(new Room(R.drawable.bogibek, "Bogibek Matyaqubov"));
        stories.add(new Room(R.drawable.elmurod, "Elmurod Nazirov"));
        stories.add(new Room(R.drawable.jabbor, "Jabbor"));
        stories.add(new Room(R.drawable.ogabekdev, "Ogabek Matyakubov"));
        stories.add(new Room(R.drawable.shakhriyor, "Shakhriyor"));
        stories.add(new Room(R.drawable.yahyo, "Yahyo Mahmudiy"));
        stories.add(new Room(R.drawable.aziz, "Azizbek"));
        stories.add(new Room(R.drawable.bogibek, "Bogibek Matyaqubov"));
        stories.add(new Room(R.drawable.elmurod, "Elmurod Nazirov"));
        stories.add(new Room(R.drawable.jabbor, "Jabbor"));
        stories.add(new Room(R.drawable.ogabekdev, "Ogabek Matyakubov"));
        stories.add(new Room(R.drawable.shakhriyor, "Shakhriyor"));
        stories.add(new Room(R.drawable.yahyo, "Yahyo Mahmudiy"));
        stories.add(new Room(R.drawable.aziz, "Azizbek"));
        stories.add(new Room(R.drawable.bogibek, "Bogibek Matyaqubov"));
        stories.add(new Room(R.drawable.elmurod, "Elmurod Nazirov"));
        stories.add(new Room(R.drawable.jabbor, "Jabbor"));
        stories.add(new Room(R.drawable.ogabekdev, "Ogabek Matyakubov"));
        stories.add(new Room(R.drawable.shakhriyor, "Shakhriyor"));
        stories.add(new Room(R.drawable.yahyo, "Yahyo Mahmudiy"));
        stories.add(new Room(R.drawable.aziz, "Azizbek"));
        stories.add(new Room(R.drawable.bogibek, "Bogibek Matyaqubov"));
        stories.add(new Room(R.drawable.elmurod, "Elmurod Nazirov"));
        stories.add(new Room(R.drawable.jabbor, "Jabbor"));
        stories.add(new Room(R.drawable.ogabekdev, "Ogabek Matyakubov"));
        stories.add(new Room(R.drawable.shakhriyor, "Shakhriyor"));
        stories.add(new Room(R.drawable.yahyo, "Yahyo Mahmudiy"));

        ArrayList<Chat> chats = new ArrayList<>();
        chats.add(new Chat(stories));
        chats.add(new Chat(new Message(R.drawable.aziz, "Azizbek")));
        chats.add(new Chat(new Message(R.drawable.bogibek, "Bogibek Matyaqubov", true)));
        chats.add(new Chat(new Message(R.drawable.elmurod, "Elmurod Nazirov", true)));
        chats.add(new Chat(new Message(R.drawable.jabbor, "Jabbor")));
        chats.add(new Chat(new Message(R.drawable.ogabekdev, "Ogabek Matyakubov", true)));
        chats.add(new Chat(new Message(R.drawable.shakhriyor, "Shakhriyor")));
        chats.add(new Chat(new Message(R.drawable.yahyo, "Yahyo Mahmudiy", true)));
        chats.add(new Chat(new Message(R.drawable.aziz, "Azizbek")));
        chats.add(new Chat(new Message(R.drawable.bogibek, "Bogibek Matyaqubov", true)));
        chats.add(new Chat(new Message(R.drawable.elmurod, "Elmurod Nazirov", true)));
        chats.add(new Chat(new Message(R.drawable.jabbor, "Jabbor")));
        chats.add(new Chat(new Message(R.drawable.ogabekdev, "Ogabek Matyakubov", true)));
        chats.add(new Chat(new Message(R.drawable.shakhriyor, "Shakhriyor")));
        chats.add(new Chat(new Message(R.drawable.yahyo, "Yahyo Mahmudiy", true)));
        chats.add(new Chat(new Message(R.drawable.aziz, "Azizbek")));
        chats.add(new Chat(new Message(R.drawable.bogibek, "Bogibek Matyaqubov", true)));
        chats.add(new Chat(new Message(R.drawable.elmurod, "Elmurod Nazirov", true)));
        chats.add(new Chat(new Message(R.drawable.jabbor, "Jabbor")));
        chats.add(new Chat(new Message(R.drawable.ogabekdev, "Ogabek Matyakubov", true)));
        chats.add(new Chat(new Message(R.drawable.shakhriyor, "Shakhriyor")));
        chats.add(new Chat(new Message(R.drawable.yahyo, "Yahyo Mahmudiy", true)));
        chats.add(new Chat(new Message(R.drawable.aziz, "Azizbek")));
        chats.add(new Chat(new Message(R.drawable.bogibek, "Bogibek Matyaqubov", true)));
        chats.add(new Chat(new Message(R.drawable.elmurod, "Elmurod Nazirov", true)));
        chats.add(new Chat(new Message(R.drawable.jabbor, "Jabbor")));
        chats.add(new Chat(new Message(R.drawable.ogabekdev, "Ogabek Matyakubov", true)));
        chats.add(new Chat(new Message(R.drawable.shakhriyor, "Shakhriyor")));
        chats.add(new Chat(new Message(R.drawable.yahyo, "Yahyo Mdiy", true)));
        chats.add(new Chat(new Message(R.drawable.aziz, "Azizbek")));
        chats.add(new Chat(new Message(R.drawable.bogibek, "Bogibek Matyaqubov", true)));
        chats.add(new Chat(new Message(R.drawable.elmurod, "Elmurod Nazirov", true)));
        chats.add(new Chat(new Message(R.drawable.jabbor, "Jabbor")));
        chats.add(new Chat(new Message(R.drawable.ogabekdev, "Ogabek Matyakubov", true)));
        chats.add(new Chat(new Message(R.drawable.shakhriyor, "Shakhriyor")));
        chats.add(new Chat(new Message(R.drawable.yahyo, "Yahyo Mnew ahmudiy", true)));

        return chats;
    }
}

File: Kotlin/app/src/main/java/dev/ogabek/kotlin/activity/MainActivity.kt
package dev.ogabek.kotlin.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import dev.ogabek.kotlin.R
import dev.ogabek.kotlin.adapter.ChatAdapter
import dev.ogabek.kotlin.model.Chat
import dev.ogabek.kotlin.model.Message
import dev.ogabek.kotlin.model.Room

class MainActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initViews()
    }

    private fun initViews() {
        recyclerView = findViewById(R.id.rv_main)
        recyclerView.layoutManager = GridLayoutManager(this, 1)
        refreshAdapter(getAllChats())
    }

    private fun refreshAdapter(chats: List<Chat>) {
        val adapter = ChatAdapter(this, getAllChats())
        recyclerView.adapter = adapter
    }

    private fun getAllChats(): ArrayList<Chat> {
        val stories: ArrayList<Room> = ArrayList<Room>()
        stories.add(Room(R.drawable.aziz, "Azizbek"))
        stories.add(Room(R.drawable.bogibek, "Bogibek Matyaqubov"))
        stories.add(Room(R.drawable.elmurod, "Elmurod Nazirov"))
        stories.add(Room(R.drawable.jabbor, "Jabbor"))
        stories.add(Room(R.drawable.ogabekdev, "Ogabek Matyakubov"))
        stories.add(Room(R.drawable.shakhriyor, "Shakhriyor"))
        stories.add(Room(R.drawable.yahyo, "Yahyo Mahmudiy"))
        stories.add(Room(R.drawable.aziz, "Azizbek"))
        stories.add(Room(R.drawable.bogibek, "Bogibek Matyaqubov"))
        stories.add(Room(R.drawable.elmurod, "Elmurod Nazirov"))
        stories.add(Room(R.drawable.jabbor, "Jabbor"))
        stories.add(Room(R.drawable.ogabekdev, "Ogabek Matyakubov"))
        stories.add(Room(R.drawable.shakhriyor, "Shakhriyor"))
        stories.add(Room(R.drawable.yahyo, "Yahyo Mahmudiy"))
        stories.add(Room(R.drawable.aziz, "Azizbek"))
        stories.add(Room(R.drawable.bogibek, "Bogibek Matyaqubov"))
        stories.add(Room(R.drawable.elmurod, "Elmurod Nazirov"))
        stories.add(Room(R.drawable.jabbor, "Jabbor"))
        stories.add(Room(R.drawable.ogabekdev, "Ogabek Matyakubov"))
        stories.add(Room(R.drawable.shakhriyor, "Shakhriyor"))
        stories.add(Room(R.drawable.yahyo, "Yahyo Mahmudiy"))
        stories.add(Room(R.drawable.aziz, "Azizbek"))
        stories.add(Room(R.drawable.bogibek, "Bogibek Matyaqubov"))
        stories.add(Room(R.drawable.elmurod, "Elmurod Nazirov"))
        stories.add(Room(R.drawable.jabbor, "Jabbor"))
        stories.add(Room(R.drawable.ogabekdev, "Ogabek Matyakubov"))
        stories.add(Room(R.drawable.shakhriyor, "Shakhriyor"))
        stories.add(Room(R.drawable.yahyo, "Yahyo Mahmudiy"))
        val chats: ArrayList<Chat> = ArrayList<Chat>()
        chats.add(Chat(stories))
        chats.add(Chat(Message(R.drawable.aziz, "Azizbek")))
        chats.add(Chat(Message(R.drawable.bogibek, "Bogibek Matyaqubov", true)))
        chats.add(Chat(Message(R.drawable.elmurod, "Elmurod Nazirov", true)))
        chats.add(Chat(Message(R.drawable.jabbor, "Jabbor")))
        chats.add(Chat(Message(R.drawable.ogabekdev, "Ogabek Matyakubov", true)))
        chats.add(Chat(Message(R.drawable.shakhriyor, "Shakhriyor")))
        chats.add(Chat(Message(R.drawable.yahyo, "Yahyo Mahmudiy", true)))
        chats.add(Chat(Message(R.drawable.aziz, "Azizbek")))
        chats.add(Chat(Message(R.drawable.bogibek, "Bogibek Matyaqubov", true)))
        chats.add(Chat(Message(R.drawable.elmurod, "Elmurod Nazirov", true)))
        chats.add(Chat(Message(R.drawable.jabbor, "Jabbor")))
        chats.add(Chat(Message(R.drawable.ogabekdev, "Ogabek Matyakubov", true)))
        chats.add(Chat(Message(R.drawable.shakhriyor, "Shakhriyor")))
        chats.add(Chat(Message(R.drawable.yahyo, "Yahyo Mahmudiy", true)))
        chats.add(Chat(Message(R.drawable.aziz, "Azizbek")))
        chats.add(Chat(Message(R.drawable.bogibek, "Bogibek Matyaqubov", true)))
        chats.add(Chat(Message(R.drawable.elmurod, "Elmurod Nazirov", true)))
        chats.add(Chat(Message(R.drawable.jabbor, "Jabbor")))
        chats.add(Chat(Message(R.drawable.ogabekdev, "Ogabek Matyakubov", true)))
        chats.add(Chat(Message(R.drawable.shakhriyor, "Shakhriyor")))
        chats.add(Chat(Message(R.drawable.yahyo, "Yahyo Mahmudiy", true)))
        chats.add(Chat(Message(R.drawable.aziz, "Azizbek")))
        chats.add(Chat(Message(R.drawable.bogibek, "Bogibek Matyaqubov", true)))
        chats.add(Chat(Message(R.drawable.elmurod, "Elmurod Nazirov", true)))
        chats.add(Chat(Message(R.drawable.jabbor, "Jabbor")))
        chats.add(Chat(Message(R.drawable.ogabekdev, "Ogabek Matyakubov", true)))
        chats.add(Chat(Message(R.drawable.shakhriyor, "Shakhriyor")))
        chats.add(Chat(Message(R.drawable.yahyo, "Yahyo Mdiy", true)))
        chats.add(Chat(Message(R.drawable.aziz, "Azizbek")))
        chats.add(Chat(Message(R.drawable.bogibek, "Bogibek Matyaqubov", true)))
        chats.add(Chat(Message(R.drawable.elmurod, "Elmurod Nazirov", true)))
        chats.add(Chat(Message(R.drawable.jabbor, "Jabbor")))
        chats.add(Chat(Message(R.drawable.ogabekdev, "Ogabek Matyakubov", true)))
        chats.add(Chat(Message(R.drawable.shakhriyor, "Shakhriyor")))
        chats.add(Chat(Message(R.drawable.yahyo, "Yahyo Mnew ahmudiy", true)))
        return chats
    }
}